From 03d4e158c6174cc05ab6288c7fde4b9c2f6dc3ba Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 2 Apr 2017 14:56:05 +0100
Subject: [PATCH] util: Stop being a complete spanner

This code decides to use various CLI programs (instead of, yknow, libraries)
with expectations of them being in the standard path. This path underps that
and makes sure /sbin/ip is always available.

Sidenote, highly inefficient code is highly inefficient.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/gpodder/util.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/gpodder/util.py b/src/gpodder/util.py
index 8e38e3d..f1d1a90 100644
--- a/src/gpodder/util.py
+++ b/src/gpodder/util.py
@@ -1802,7 +1802,7 @@ def linux_get_active_interfaces():
     empty list if the device is offline. The loopback
     interface is not included.
     """
-    process = subprocess.Popen(['ip', 'link'], stdout=subprocess.PIPE)
+    process = subprocess.Popen(['/sbin/ip', 'link'], stdout=subprocess.PIPE)
     data, _ = process.communicate()
     for interface, _ in re.findall(r'\d+: ([^:]+):.*state (UP|UNKNOWN)', data):
         if interface != 'lo':
@@ -1862,7 +1862,7 @@ def connection_available():
                     offline = True
 
             # If we assume we're offline, try the "ip" command as fallback
-            if offline and find_command('ip') is not None:
+            if offline:
                 if len(list(linux_get_active_interfaces())) == 0:
                     offline = True
                 else:
-- 
2.12.2

